src/Data/Interval.hs

```diff
@@ -115,48 +116,22 @@ infix 4 >??
 infix 4 /=??
 
 -- | The intervals (/i.e./ connected and convex subsets) over real numbers __R__.
-data Interval r = Interval !(Extended r, Bool) !(Extended r, Bool)
-  deriving (Eq, Typeable)
+data Interval r = Interval
+  { -- | 'lowerBound' of the interval and whether it is included in the interval.
+    -- The result is convenient to use as an argument for 'interval'.
+    lowerBound' :: !(Extended r, Bool)
+  , -- | 'upperBound' of the interval and whether it is included in the interval.
+    -- The result is convenient to use as an argument for 'interval'.
+    upperBound' :: !(Extended r, Bool)
+  } deriving (Eq, Generic, Data, Typeable)
```

Comment:
This PR replaces hand-written `Data` instance with the one generated by GHC.
But doesn’t the generated one allow constructing unnormalized interval value (e.g. through `gmapT`)?

Reply:
I confirmed that following `y` becomes unnormalized interval where lower bounds is `Finite (-1)` while upper bound is `Finite 0`.

```
{-# LANGUAGE ScopedTypeVariables #-}
import qualified Data.Interval as Interval
import Data.Data
import Data.Maybe

f :: Data a => a -> a
f x = fromJust $ do
  (x' :: Interval.Extended Integer, b::Bool) <- cast x
  cast (fmap negate x', b)

x :: Interval.Interval Integer
x = 0 Interval.<=..<= 1

y :: Interval.Interval Integer
y = gmapT f x
```

So could you revert this change?

Reply:
Done, my bad.